sunscreen_compiler_common/src/context.rs

```diff
@@ -278,15 +364,49 @@ where
     pub fn add_node(&mut self, operation: O) -> NodeIndex {
         #[cfg(feature = "debugger")]
         {
-            let group_id = self.group_counter;
+            // Capture backtrace and insert into lookup
+            let bt = Backtrace::new();
+            let stack_frames = self
+                .graph
+                .metadata
+                .stack_lookup
+                .backtrace_to_stackframes(bt);
+            let serialized_stack_frames = stack_frames
+                .iter()
+                .map(|frame| frame.serialize())
+                .collect::<Vec<_>>()
+                .join("_");
+
+            let stack_id = *self
+                .graph
+                .metadata
+                .stack_lookup
+                .data_id_lookup
+                .entry(serialized_stack_frames.clone())
+                .or_insert_with(|| {
+                    self.graph.metadata.stack_counter += 1;
+                    self.graph.metadata.stack_counter
+                });
+
+            let group_id = self.graph.metadata.group_counter;
+
+            self.graph
+                .metadata
+                .stack_lookup
+                .id_data_lookup
```

Comment:
I'd make methods on the Lookup object for `id_for_data` and `data_for_id` so these HashMaps aren't public.

That way if we change to a trie or something, the interface is the same.